ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/recovery/BasicXARecovery.java

```diff
@@ -202,7 +204,9 @@ private final JDBC2RecoveryConnection getConnection ()
 			if (dc != null)
 				dbProperties.put(TransactionalDriver.dynamicClass, dc);
 
-			return new JDBC2RecoveryConnection(theURL, dbProperties);
+			JDBC2RecoveryConnection connection = new JDBC2RecoveryConnection(theURL, dbProperties);
+			connections.add(connection);
```

Comment:
So the GC will clean up the connections when BasicXARecovery is GC'ed (it might be worth putting that in a code comment somewhere). Are you certain that there isn't a memory leak here (how long are instances of BasicXARecovery held for)?